File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    w = []
    h = []
    for i, j in sizes:
        if i < j:
            i, j = j, i  # 가로보다 세로가 긴 경우 세로를 가로로 넣는다.
        w.append(i)
        h.append(j)

    return max(w) * max(h)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int w = 0, h = 0;
        for (int[] card : sizes) {
            w = Math.max(w, Math.max(card[0], card[1]));
            h = Math.max(h, Math.min(card[0], card[1]));
        }
        int answer = w * h;
        return answer;
    }
}